rellic/AST/Z3ConvVisitor.cpp

```diff
@@ -564,18 +580,34 @@ bool Z3ConvVisitor::VisitBinaryOperator(clang::BinaryOperator *c_op) {
       InsertZ3Expr(c_op, lhs & rhs);
       break;
 
+    case clang::BO_Or:
+      InsertZ3Expr(c_op, lhs | rhs);
+      break;
+
     case clang::BO_Xor:
       InsertZ3Expr(c_op, lhs ^ rhs);
       break;
 
+    case clang::BO_Mul:
+      InsertZ3Expr(c_op, lhs * rhs);
+      break;
+
+    case clang::BO_Div:
+      InsertZ3Expr(c_op, lhs / rhs);
+      break;
+
     case clang::BO_Shr:
       InsertZ3Expr(c_op, c_op->getLHS()->getType()->isSignedIntegerType()
                              ? z3::ashr(lhs, rhs)
                              : z3::lshr(lhs, rhs));
       break;
 
+    case clang::BO_Shl:
+      InsertZ3Expr(c_op, z3::shl(lhs, rhs));
```

Comment:
Does this make sense?

Reply:
Yeah, I think I would've written the same ;)